compiler-rt/lib/asan/asan_poisoning.cpp

```diff
@@ -20,11 +20,33 @@
 #include "sanitizer_common/sanitizer_flags.h"
 #include "sanitizer_common/sanitizer_interface_internal.h"
 #include "sanitizer_common/sanitizer_libc.h"
+#include "sanitizer_common/sanitizer_stackdepot.h"
 
 namespace __asan {
 
 static atomic_uint8_t can_poison_memory;
 
+static Mutex PoisonRecordsMutex;
+static PoisonRecordRingBuffer *PoisonRecords = nullptr;
+
+void InitializePoisonTracking() {
```

Comment:
we have lock, lazy init under the lock to have less code?

Reply:
Done.